tsv_parser.go

```
@@ -0,0 +1,75 @@
+package iosupport
+
+import (
+	"bytes"
+)
+
+// UnescapeSeparator cleans composed separator like \t
+func UnescapeSeparator(separator string) byte {
+	if separator == "\\t" {
+		separator = "\t"
+	}
+	return []byte(separator)[0]
+}
+
+// TrimNewline removes newline characters at the end of line
+func TrimNewline(line []byte) []byte {
+	line = bytes.TrimRight(line, "\n")
+	line = bytes.TrimRight(line, "\r")
+	return bytes.TrimRight(line, "\n")
+}
+
+// A TsvParser reads records from a TSV-encoded file.
+type TsvParser struct {
+	*Scanner
+	err       error // Sticky error.
```

The error is not returned by the called method as usual.